### PR TITLE
Refactor unit tests: move test data to .csv files

### PR DESCRIPTION
**Pull request by dsuponitskiy**

### CODE REVIEW

File: src/pke/include/constants.h [hunk 1]
```diff
@@ -37,6 +37,10 @@
 // TODO: Review whether we need to include hal.h.
```
Comment (jbates-duality) [inline src/pke/include/constants.h:37]: Agree with this TODO.  I am envisioning that this file should be as lightweight as possible...

File: src/pke/include/scheme/cryptocontextparams-base.h [hunk 2]
```diff
@@ -167,14 +169,123 @@ class Params {
         SetToDefaults(scheme0);
     }
 
+    /**
+     * This Params' constructor is to be used for unittests only.
+     *
+     * @param vals - vector with override values. sequence of vals' elements must be the same as we get it from getAllParamsDataMembers()
+     * @return a vector with names of all data members of Params
+     */
+    explicit Params(const std::vector<std::string>& vals) {
```
Comment (jbates-duality) [inline src/pke/include/scheme/cryptocontextparams-base.h:178]: This is kind of a huge method for an implicit inline.  I understand that it's not for general-purpose use.  Does it make sense to move this from a header to an implementation file?

**Review by jbates-duality — COMMENTED** [review]
See comments in code.

File: src/core/lib/lattice/stdlatticeparms.cpp [hunk 3]
```diff
@@ -38,6 +38,20 @@
 
 namespace lbcrypto {
 
+SecurityLevel convertToSecurityLevel(const std::string& str) {
```
[the remark this answers is not in the document]
Comment (dsuponitskiy) [reply src/core/lib/lattice/stdlatticeparms.cpp:41]: My approach is simple: use unordered_map if the list of enums is more than 10 elements. However, I used unordered_map in unit tests as I expected the number of values to convert to grow.

File: src/pke/include/scheme/cryptocontextparams-base.h (hunk 2 above)
Comment (dsuponitskiy) [reply src/pke/include/scheme/cryptocontextparams-base.h:178]: Agree. Will change.

File: src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp [hunk 4]
```diff
@@ -28,44 +28,51 @@
...
 
 //===========================================================================================================
 enum TEST_CASE_TYPE {
     BGVRNS_AUTOMORPHISM = 0,
     EVAL_AT_INDX_PACKED_ARRAY,
     EVAL_SUM_PACKED_ARRAY,
 };
-
+TEST_CASE_TYPE convertStringToCaseType(const std::string& str) {
+    const std::unordered_map<std::string, TEST_CASE_TYPE> stringToCaseType = {
```
[the remark this answers is not in the document]
Comment (dsuponitskiy) [reply src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp:54]: The map is used in the conversion function only and I don't think that the change would improve performance or lessen the memory footprint.

File: src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp [hunk 5]
```diff
@@ -28,44 +28,51 @@
...
-        case EVAL_AT_INDX_PACKED_ARRAY:
-            typeName = "EVAL_AT_INDX_PACKED_ARRAY";
-            break;
-        case EVAL_SUM_PACKED_ARRAY:
-            typeName = "EVAL_SUM_PACKED_ARRAY";
-            break;
-        default:
-            typeName = "UNKNOWN_UTBGVRNS_AUTOMORPHISM";
-            break;
+    const std::unordered_map<TEST_CASE_TYPE, std::string> caseTypeToString = {
```
[the remark this answers is not in the document]
Comment (dsuponitskiy) [reply src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp:65]: The map is used in the conversion function only and I don't think that the change would improve performance or lessen the memory footprint.

File: src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp [hunk 6]
```diff
@@ -77,17 +84,33 @@ enum TEST_CASE_ERROR {
     INVALID_BATCH_SIZE,
     NO_KEY_GEN_CALL
 };
-
-struct TEST_CASE_UTBGVRNS_AUTOMORPHISM {
+TEST_CASE_ERROR convertStringToCaseError(const std::string& str) {
```
[the remark this answers is not in the document]
Comment (dsuponitskiy) [reply src/pke/unittest/utbgvrns/UnitTestBGVrnsAutomorphism.cpp:87]: The map is used in the conversion function only and I don't think that the change would improve performance or lessen the memory footprint.

File: src/pke/include/scheme/cryptocontextparams-base.h (hunk 2 above)
Comment (jbates-duality) [reply src/pke/include/scheme/cryptocontextparams-base.h]: i see this has been moved to the implementation file.  thank you.

**Review by jbates-duality — APPROVED** [review]
All comments addressed.  Approving.

**Pull request merged**